server/graphql/v2/interface/Activity.ts

```diff
@@ -0,0 +1,44 @@
+import { GraphQLString, GraphQLNonNull, GraphQLObjectType } from 'graphql';
+import { GraphQLDateTime } from 'graphql-iso-date';
+import { ActivityType } from '../enum';
+import { Account } from './Account';
+import { Individual } from '../object/Individual';
+import { getIdEncodeResolver, IDENTIFIER_TYPES } from '../identifiers';
+
+export const Activity = new GraphQLObjectType({
+  name: 'Activity',
+  description: 'An activity describing something that happened on the platform',
+  fields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: 'Unique identifier for this activity',
+      resolve: getIdEncodeResolver(IDENTIFIER_TYPES.ACTIVITY),
+    },
+    type: {
+      type: new GraphQLNonNull(ActivityType),
+      description: 'The type of the activity',
+    },
+    createdAt: {
+      type: new GraphQLNonNull(GraphQLDateTime),
+      description: 'The date on which the ConnectedAccount was created',
+    },
+    account: {
+      type: Account,
+      description: 'The account concerned by this activity, if any',
+      resolve: async (activity, _, req): Promise<object> => {
+        if (activity.CollectiveId) {
+          return req.loaders.Collective.byId.load(activity.CollectiveId);
+        }
+      },
+    },
+    individual: {
```

Comment:
Maybe we should rename this to `fromIndividual` for more clarity

Reply:
I think we don't need "from" because there's no "to" in an activity.
Perhaps we should keep naming it Account for consistency?

Reply:
I think we don't need "from" because there's no "to" in an activity.
Perhaps we should keep naming it Account for consistency?

Reply:
There's already an `account` field. Also this one will always be an individual, it cannot be a collective, an org or an event.

But to you mean `individualAccount`?

Reply:
For the case where an individual is taking action in the name of a collective?
Interesting, I think it is a good idea to bind the data-type. Meaning that if both Individual and Account are returning the same data type, they should have similar names, specially because we don't use TS in the frontend.
I think `individualAccount` could be a nice solution.

Reply:
> For the case where an individual is taking action in the name of a collective?

Even then, this endpoint will return the user that triggered the action (we rely on the `UserId` field). The goal in such case is to know which user triggered the action in the name of the collective.


> Meaning that if both Individual and Account are returning the same data type

It's not exactly the case, the endpoint returns an `Individual` object,  which inherits of the `Account` interface but has its own type.

Reply:
Nice, thanks for explaining it to me. Perhaps Individual is enough then, sorry for overcomplicating.

Reply:
No over-complication :slightly_smiling_face: Asking for more clarity is always good, no matter if it results in a code change or not!